homeassistant/components/zwave_js/climate.py

```diff
@@ -479,3 +483,42 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
         if preset_mode_value is None:
             raise ValueError(f"Received an invalid preset mode: {preset_mode}")
         await self.info.node.async_set_value(self._current_mode, preset_mode_value)
+
+
+class DynamicCurrentTempClimate(ZWaveClimate):
+    """Representation of a thermostat that can dynamically use a different Zwave Value for current temp."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize thermostat."""
+        super().__init__(config_entry, client, info)
+        self._current_temp_dependent_value = None
+        if not self.info.platform_helpers:
+            return
+
+        # Lookup map to map the transformed value of a dependent Value to a current
+        # temperature Value
+        self._lookup = {}
+        lookup = self.info.platform_helpers["lookup"]
+        for key in lookup:
```

Comment:
The compiling here belongs in the discovery data class for the problem case, I think.